src/devices/bus/hp_dio/hp98603.cpp

```diff
@@ -0,0 +1,74 @@
+// license:BSD-3-Clause
+// copyright-holders:Sven Schnelle
+/***************************************************************************
+
+  HP98603 BASIC ROM card
+
+***************************************************************************/
+
+#include "emu.h"
+#include "hp98603.h"
+#include "screen.h"
+#include "softlist.h"
+#include "bus/generic/slot.h"
+#include "bus/generic/carts.h"
+
+DEFINE_DEVICE_TYPE(HPDIO_98603, dio16_98603_device, "dio98603", "HP98603 BASIC ROM card")
+
+MACHINE_CONFIG_MEMBER( dio16_98603_device::device_add_mconfig )
+        MCFG_GENERIC_CARTSLOT_ADD("cartslot", generic_plain_slot, "98603_cart")
+
+        MCFG_GENERIC_EXTENSIONS("bin,rom")
+        MCFG_GENERIC_LOAD(dio16_98603_device, hp98603_rom_slot)
+	MCFG_SOFTWARE_LIST_ADD("cart_list", "98603")
+MACHINE_CONFIG_END
+
+const tiny_rom_entry *dio16_98603_device::device_rom_region() const
+{
+	return NULL;
```

Comment:
could you please use the magic `nullptr` rather than the macro `NULL` as a generic null pointer value?  Also, there's no reason to override `device_rom_region` if the card has no fixed ROMs.